site/topics/lists/alias-trivia.rst

```diff
@@ -79,82 +79,65 @@ Aliases & List Trivia
-
-Side effects
-============
+    * The parameter within the function will be an alias
 
+.. code-block:: python
+    :linenos:
 
-* Consider the code::
+    def add_to_list(some_list, value):
+        some_list.append(value)
 
-    def add_to_list(my_list):
-        my_list.append('appended')
+    a_list = ['a', 'b', 'c']
+    add_to_list(a_list, 99)
+    print(a_list)   # Results in ['a', 'b', 'c', 99]
 
-* Now consider the code::
 
-    def add_to_list_2(my_list):
-        return my_list + ['appended']
+* In the above example, although never access through ``a_list``, the list ``a_list`` references is altered through the alias ``some_list`` within the function ``add_to_list``
 
-.. admonition:: Activity
-    :class: activity
 
-    What happens when you do this?
+Side Effects & Pure Functions
+-----------------------------
 
-        >>> a = [1,2,3,4]
-        >>> add_to_list(a)
-        >>> print(a)
+* ``add_to_list`` is an example of a function that has a *side effect*
 
-   How about this:
+    * The function modified the list that was passed by reference
+    * The term *side effect* comes from our mathematical expectation of a *function*
 
-        >>> a = [1,2,3,4]
-        >>> add_to_list_2(a)
-        >>> print(a)
+        * A function maps some parameters on to a value
+        * If I give you the function :math:`f(x, y, z)= x + y - z` and ask you to evaluate :math:`f(1, 2, 3)`, you don't expect the values of :math:`x`, :math:`y`, and :math:`z` to change
 
-   Finally, how about this:
+* We can write a different version of the function that has no side effect
 
-        >>> a = [1,2,3,4]
-        >>> b = add_to_list_2(a)
-        >>> print(a)
-        >>> print(b)
+    * Functions without side effects are called *pure* functions
 
-* The function ``add_to_list`` *modified* the parameter you passed in.
-* The function ``add_to_list_2`` kept a respectful distance from your parameter and, instead, created a *new* list and *returned* that as the answer.
-* If a function modifies a parameter it is said to have *side effects*.
-    * The term "side effect" comes from our mathematical expectation of a "function". A function maps some parameters on to a value. If I give you the function `f(x,y,z)=x+y-z` and ask you to evaluate `f(1,2,3)`, you don't expect the values of `x`, `y` and `z` to change!
+.. code-block:: python
+    :linenos:
 
-Pure functions
-==============
-* If a function has no side effects, we call it a *pure function*.
-* Some programming languages allow *only* pure functions (e.g., `Haskell <http://www.haskell.org/haskellwiki/Haskell>`_).
-    * There are some nice theoretical, and practical benefits to this.
-* As you might guess from the ameliorative term "pure"... functions with side effects are considered... "not pure"... even downright dirty, by some folks.
+    def add_to_list_pure(some_list, value):
+        new_list = some_list.copy()
+        new_list.append(value)
+        return new_list
```

Comment:
`return some_list + [value]` ? or just being super explicit about the copy being created?

Reply:
> `return some_list + [value]` ? 

Originally had this 

> or just being super explicit about the copy being created?

Exactly